stellar-contract-env-host/src/host.rs

```diff
@@ -389,6 +396,38 @@ impl Host {
         // Resolve the function symbol and invoke contract call
         vm.invoke_function_raw(self, SymbolStr::from(func).as_ref(), args)
     }
+
+    pub fn invoke_function(&mut self, hf: HostFunction, args: ScVec) -> Result<ScVal, HostError> {
+        let contract_id = match &args.as_vec()[0] {
```

Comment:
❓ I noticed in this PR you go to some length to introduce new error handling that reduces the panicking in an existing function, however I also see `[]` in use here on a `Vec` which will panic if the vec is short. Is the panicking here intended?